### PR TITLE
Added Clara Client and Types

### PR DESCRIPTION
**Pull request by KavinKrishnan**
The Clara Client allows users to request GPU utilization details, Platform version information, and the stoppage of Pipeline Service and Triton

### CODE REVIEW

File: nvidia_clara/model_types.py [hunk 1]
```diff
@@ -139,12 +139,12 @@ def name(self, name: str = None):
...
 
     @property
-    def tags(self) -> HashMap[str, str]:
+    def tags(self) -> Mapping[str, str]:
         """The set of tags / meta-data associated with this infrence model."""
         return self._tags
 
     @tags.setter
-    def tags(self, tags: HashMap[str, str] = None):
+    def tags(self, tags: Mapping[str, str] = None):
```
Comment (whoisj) [inline nvidia_clara/model_types.py:147]: `Mapping` vs `HashMap`, why the change?

File: nvidia_clara/pipeline_types.py [hunk 2]
```diff
@@ -162,7 +162,7 @@ def metadata(self, metadata: Mapping[str, str]):
 
 class PipelineInfo:
 
-    def __init__(self, pipeline_id: PipelineId = None, name: str = None, metadata: Mapping[str, str] = None):
+    def __init__(self, pipeline_id: PipelineId = None, name: str = None, metadata: Mapping[str, str] = dict()):
```
Comment (whoisj) [inline nvidia_clara/pipeline_types.py:165]: you understand that `dict()` in place of `None` requires an allocation, that you won't likely use, to be made - right?

Comment (KavinKrishnan) [reply nvidia_clara/pipeline_types.py:165]: The motivation is to allow users to be able to easily append to the metadata variable of an instance even after not specifying it in the constructor

File: nvidia_clara/model_types.py (hunk 1 above)
Comment (KavinKrishnan) [reply nvidia_clara/model_types.py:147]: The python map/dictionary is a hash map, there is no actual HashMap class. Not sure how it was working before, but according to the Python official documentation using `Mapping` seems to be the proper convention.

**Review by whoisj — APPROVED** [review]
